payjoin-relay/src/db.rs

```diff
@@ -1,189 +1,355 @@
-use std::time::Duration;
+#![allow(unused_imports)]
+use std::{
+    fmt::Display,
+    pin::Pin,
+    sync::{Arc, Mutex},
+    thread,
+    time::Duration,
+};
 
 use anyhow::Result;
-use sqlx::postgres::{PgListener, PgPoolOptions};
-use sqlx::{PgPool, Pool, Postgres};
-use tracing::debug;
+use futures_util::StreamExt;
+use redis::{
+    aio::{self, AsyncStream},
+    AsyncCommands, Client, Commands, Connection, ControlFlow, PubSub, PubSubCommands, RedisError,
+};
 
-const RES_COLUMN: &str = "res";
-const REQ_COLUMN: &str = "req";
+pub type DatabaseClient = Arc<Mutex<Client>>;
 
-pub(crate) struct DbPool {
-    pool: Pool<Postgres>,
-    timeout: Duration,
+pub trait RedisPool {
+    fn client(&self) -> &DatabaseClient;
```

Comment:
The point of this abstraction is to hide the details of the database client to only the exact needs of the application. Exposing the client is a good sign things are going off the tracks